### ISSUE TITLE
How to properly set SmartTab approach config?

### ISSUE THREAD

**Issue by maximal** [squizlabs/PHP_CodeSniffer#2480]
I want to have strict SmartTab codestyle. Let’s call it [PSR-2T](https://github.com/maximal/psr-2t) (PSR-2 with SmartTab).

I use this config to set it up:
![image](https://user-images.githubusercontent.com/980679/56129085-11c8ab00-5f8a-11e9-8932-348d9d1f5a06.png)

Also I have an alignment by HTML-attributes in PHP templates (with spaces, of course).
But CodeSniffer yield an error where it should not:
![image](https://user-images.githubusercontent.com/980679/56129110-20af5d80-5f8a-11e9-976d-8093d7c9173d.png)
Here i should have exact 5 (no more, no less) tabs at the beginning of the line and any amount of spaces after the tab sequence.

How can I provide a valid config for SmartTab approach: indenting with tabs, but aligmnent (if present) with spaces?

My test method:
```php
	public function checkSmartTabs()
	{
		$long_1 = 'long1';       // VALID (ok)
		$long_2 = 'long2';       // VALID (ok)
		    $s1 = 'short1';      // SHOULD NOT BE AN ERROR BUT IT IS
		    $s2 = 'short2';      // SHOULD NOT BE AN ERROR BUT IT IS
			$s3 = 'short3';      // SHOULD BE AN ERROR BUT IT IS NOT
			$s4 = 'short4';      // SHOULD BE AN ERROR BUT IT IS NOT
	    	$s5 = 'short5';      // ERROR (ok)

		if (true) {
		  // Comment             // SHOULD BE AN ERROR BUT IT IS NOT
			$s6 = 'short6';      // VALID (ok)
				$s7 = 'short7';  // SHOULD BE AN ERROR BUT IT IS NOT
		}
	}
```

![image](https://user-images.githubusercontent.com/980679/56129992-5c4b2700-5f8c-11e9-9c12-8b399a962e8b.png)

All errors here are: ![image](https://user-images.githubusercontent.com/980679/56129614-4426d800-5f8b-11e9-904a-10abc6f22163.png)

Also see #547.

**Comment by gsherwood** [comment]
> How can I provide a valid config for SmartTab approach: indenting with tabs, but aligmnent (if present) with spaces?

The built-in sniffs already allow this, but they use the tab-width setting to determine if tabs or spaces should be used.

If your tab-width is set to 4 (the default), it means that you can't indent with 5 tabs than 5 spaces. The sniffs wants 6 tabs and 1 space. The space is used for precision indents only.

Based on your examples, you want a very different indent approach, and PHPCS does not support this. You'd need to write a custom sniff to implement it.

**Comment by maximal** [comment]
@gsherwood, then, why does the sniff consider commented line (inside of `if` in my example) valid?

You should have three tabs there, aren’t you?

**Comment by gsherwood** [comment]
> @gsherwood, then, why does the sniff consider commented line (inside of `if` in my example) valid?

Because PSR-2 explicitly ignores "Comments and documentation blocks", so the error is suppressed [in the ruleset](https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Standards/PSR2/ruleset.xml#L59). If you use any other standard, you'll get an error for incorrectly indented comment lines as well.
